class-tgm-plugin-activation.php

```diff
@@ -2023,12 +2064,31 @@ public function force_activation() {
 					} elseif ( $this->can_plugin_activate( $slug ) ) {
 						// There we go, activate the plugin.
 						activate_plugin( $plugin['file_path'] );
+						$this->append_transient( 'tgmpa_force_activated_plugins', $slug );
 					}
 				}
 			}
 		}
 
 		/**
+		 * Set/update a transient, appending the new value if the transient already exists.
+		 *
+		 * @since 2.6.0
+		 *
+		 * @param string       $transient_key Name of transient.
+		 * @param string|array $data To set or add to transient.
+		 */
+		protected function append_transient( $transient_key, $data ) {
+			if ( $transient_data = get_transient( $transient_key ) ) {
+				(array) $transient_data[] = $data;
+			} else {
+				$transient_data = (array) $data;
+			}
+
+			set_transient( $transient_key, $transient_data );
+		}
```

Comment:
I think the answer to that is whether or not plugins could be deactivated over multiple requests... if yes, then we should keep as is so that the list could aggregate in the transient until it is finally displayed by the admin_notices hook. If not (we know all plugins will be deactivated on the same request), then this could be refactored to collect the plugin list and add to the transient all at once.